drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc

```diff
@@ -203,6 +203,21 @@ function boincwork_get_project_specific_config() {
   
   // Convert XML to array for validation
   $xml = load_configuration($raw_config_data);
+  // By default the returned array should have an 'app_overrides'
+  // element. Even if the admin did not specificy an <apps> container
+  // in the XML, a basic one is added. This is because applications
+  // are now queried from the database, and the content of the XML
+  // upload is an override.
+  if (!array_key_exists('app_overrides', $xml['project_specific_preferences'])) {
+    $simpleapps = array(
+      'title' => array(
+        '@value' => 'Run only the selected applications',
+        '@position' => 1,
+      ),
+      '@position' => 0,
+    );
+    $xml['project_specific_preferences']['app_overrides'] = $simpleapps;
```

Comment:
This `$xml` is used for validation, I'm going to update the schema (XSD) and make `app_overrides` mandatory, ok?